app/controllers/references_controller.rb

```diff
@@ -1,10 +1,37 @@
 class ReferencesController < ApplicationController
   before_action :ensure_current_user
 
+  # Shame: refactor the hell out of this
   def create
     list = List.find(params[:list_id])
-    paper = Paper.find(params[:reference][:paper_id])
-    Reference.create!(list_id: list.id, paper_id: paper.id)
+
+    # Check if uuid. If not, then ask Pubmed.
+    paper_id = params[:reference][:paper_id]
+    if uuid? paper_id
+      paper = Paper.find(paper_id)
+    else
+      pubmed = Pubmed.new
+      results = pubmed.search paper_id
+      data = results['result'][paper_id]
+      paper = (
+        Paper.find_by(pubmed_id: paper_id) ||
+        Paper.find_by(doi: paper_id) ||
+        # I really hate this due to long experience with it,
+        # but we should probably some sort of mapping
+        Paper.create(
+          pubmed_id: data['uid'],
+          title: data['title'],
+          published_at: data['pubdate'],
+          doi: data['elocationid'].sub(/^doi: /, "")
```

Comment:
Sure. I would, but I'm stuck on the highway.

Hurry. The zombies are coming.

On Sat, Jul 23, 2016, 11:58 PM wsf246 <notifications@github.com> wrote:

> In app/controllers/references_controller.rb
> <https://github.com/briankung/projectcredo/pull/28#discussion_r71981864>:
>
> > +    if uuid? paper_id
> > +      paper = Paper.find(paper_id)
> > +    else
> > +      pubmed = Pubmed.new
> > +      results = pubmed.search paper_id
> > +      data = results['result'][paper_id]
> > +      paper = (
> > +        Paper.find_by(pubmed_id: paper_id) ||
> > +        Paper.find_by(doi: paper_id) ||
> > +        # I really hate this due to long experience with it,
> > +        # but we should probably some sort of mapping
> > +        Paper.create(
> > +          pubmed_id: data['uid'],
> > +          title: data['title'],
> > +          published_at: data['pubdate'],
> > +          doi: data['elocationid'].sub(/^doi: /, "")
>
> I can take a stab at that?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/briankung/projectcredo/pull/28/files/f17fd0bc5346d3be6254486f0414e4c4d0e6407b#r71981864>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACtGxypCv-wUW_6G1gqmY6q-Cys37-csks5qYvDLgaJpZM4JTTYA>
> .
>